compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/draganddrop/DragAndDrop.desktop.kt

```diff
@@ -40,6 +40,13 @@ actual class DragAndDropTransferData(
     @property:ExperimentalComposeUiApi
     val supportedActions: Iterable<DragAndDropTransferAction>,
 
+    /**
+     * The initial transfer action when the drag starts; must be one of the values in
+     * [supportedActions].
+     */
+    @property:ExperimentalComposeUiApi
+    val initialAction: DragAndDropTransferAction,
```

Comment:
This feature doesn't work on Windows, the initial cursor is still Move.

I mentioned that in the initial PR.

Checked on this sample:
```
@Sampled
@Composable
fun TextDragAndDropSourceSample(modifier: Modifier) {
    val label = remember { "Drag mef" }
    Box(
        modifier =
            modifier
                .dragAndDropSource { _ ->
                    DragAndDropTransferData(
                        clipData = ClipData.newPlainText(label, label),
                        flags = View.DRAG_FLAG_GLOBAL,
                        supportedActions = listOf(DragAndDropTransferAction.Move, DragAndDropTransferAction.Link),
                        initialAction = DragAndDropTransferAction.Copy
                    )
                }
                .border(
                    border =
                        BorderStroke(
                            width = 4.dp,
                            brush = Brush.linearGradient(listOf(Color.Magenta, Color.Magenta))
                        ),
                    shape = RoundedCornerShape(16.dp)
                )
                .padding(24.dp),
    ) {
        Text(modifier = Modifier.align(Alignment.Center), text = label)
    }
}
```

It is fine if some system doesn't support it and fallbacks to the default behaviour. But we should mention that in the KDoc, and add something like `isDragAndDropInitialActionSupported()`

Reply:
> supportedActions = listOf(DragAndDropTransferAction.Move, DragAndDropTransferAction.Link),
> initialAction = DragAndDropTransferAction.Copy

`initialAction` must be one of the `supportedActions`, and in fact, the whole drag-and-drop session is cancelled if it's not.

That's what this PR fixes, actually. As the initial action was hardcoded to be `MOVE`, if `supportedActions` didn't include it as one of the values, the drag-and-drop session was cancelled by AWT immediately.

Reply:
> initialAction must be one of the supportedActions

It was mistake, during copy-paste. Fixed

Reply:
> It was a mistake during copy-paste. Fixed. Every initial icon is Move, no matter the arguments.

Possibly a bug in Java on Windows. Hard to say without digging into it.

But
1. This feature is not (only) about the icon. It's also about the semantics of the transfer. The initial action is also the final action itself, if the user doesn't change it. Using `COPY`, for example, will copy the data rather than "moving" it.
2. If it is indeed a bug, I don't think the right solution is `isDragAndDropInitialActionSupported()`. Firstly because of 1 (the action is supported I assume, only the icon isn't). But also, how do we know when a non-`move` icon is supported? Perhaps it works fine on other versions of Windows or JVM... The best we could do is document that "some systems are known to not display the correct pointer icon corresponding to the action".

Reply:
I checked drag&drop of a file from Compose to File Explorer on Windows 11 and Corretto 17/JBR 17:
```
                DragAndDropTransferData(
                    transferable = DragAndDropTransferable(
                        object : Transferable {
                            override fun getTransferDataFlavors() = arrayOf(DataFlavor.javaFileListFlavor)
                            override fun isDataFlavorSupported(flavor: DataFlavor?) = true
                            override fun getTransferData(flavor: DataFlavor?) = listOf(File("D:/1.txt"))
                        }
                    ),
                    supportedActions = listOf(
                        DragAndDropTransferAction.Copy,
                        DragAndDropTransferAction.Move,
                        DragAndDropTransferAction.Link,
                    ),
                    initialAction = DragAndDropTransferAction.Copy,
                    dragDecorationOffset = offset,
                    onTransferCompleted = { action ->
                        println("Action at source: $action")
                    }
                )
```
The results are:
1. `supportedActions` works
2. `initialAction` doesn't work semantically. The actual initial action is defined by `supportedActions`:
```
Copy, Move, Link -> Move
Move, Link, Copy -> Move
Copy , Link -> Copy
Link, Copy -> Copy
Link -> Link
```

3. It moves/copies/links the real file `D:/1.txt`. If it doesn't exist, File explorer just will show an error.

4. "Link" actually isn't supported in the default File Explorer, but supported in the third party explorer. Not sure why.

Reply:
> initialAction doesn't work semantically. The actual initial action is defined by supportedActions

I investigated it some more. The action is not defined just by `supportedActions`, but by the target app, with the constraint that it's one of the supported actions.

For example, if I drag a file to the Finder, the action is "Move", but if I drag it into Notes, the action is "Copy".
For some reason, when dragging from our `dragSource` to our `dragTarget`, the action is always "Move".

Something that surprised me is that when I dragged a "File" from a Compose app to the Finder, it actually moved the file from its original location to the folder into which I dragged it! I was expecting that it will leave the deletion of the file from its original location to the app.

So, what do we want to do here?

Reply:
If we think that the "initial action" we pass to `exportAsDrag` doesn't do anything (other than needing to pass the validation that it's one of the supported actions), then we could just pick one of the actions in `supportedActions` and pass it.

Reply:
> For some reason, when dragging from our dragSource to our dragTarget, the action is always "Move".

So, it seems not the Windows specific issue? Does it only validates the supported action in your test cases? If so then we shouldn't add API for that and do as you suggested - pick one of the `supportedActions`.

I tested more cases and also concluded that it is determined by the receiver on Windows:
1. When we drag from Compose to the default explorer, the initial action always "Copy" if it is supported
2. When we drag from Compose to the third party explorer "Directory Opus", the initial action always "Move" if it is supported
3. When we drag a file from "D:" to "C:" within the default File Explorer, the action is Copy. When we drag within one disk it is Move. When we drag it to the third-party explorer, it is determined by this setting:
<img width="451" alt="image" src="https://github.com/user-attachments/assets/4b50500e-396f-4b7a-9ddc-859f77e64ea0">
MOVEWHENTHESAME - Copy if disk is different, Move is the same
MOVE - always Move

Though I didn't find how we can defined the action for the receiver in Swing.